### PR TITLE
#8584 Add E2E tests for page-specific features

### LINKED ISSUE #8584: Live Editing - Page specific features
Page Editor e2e tests to cover that relate to edge specific page edge cases 
- **Special Page Detection**
    - Verify that a special page is shown if the target page is the extension console page, or an internal chrome page.
- **Reload Message**
    - “Reload” message is shown over the whole page editor when changes are made in the background.
        - Ex. clicking on an activation link.
- **Unavailable Mod Indicator**
    - “Not available on the page” icon in the mod listing panel when the mod would not be run for the target URL.

### PR DESCRIPTION
**Pull request by mnholtz**
## What does this PR do?

- Closes #8584
- This PR modifies the end-to-end tests and the page editor to handle restricted pages and unavailable mods.
- Note that it also broadens the tab query in `TabInspectionGate.tsx` (the component that renders the list of tabs to open in the tab-version of the Page Editor) to include restricted browser pages (in order to enable testing restricted browser page behavior in the page edtior)

### CODE REVIEW

File: src/pageEditor/context/TabInspectionGate.tsx [hunk 1]
```diff
@@ -31,13 +31,7 @@ import { getErrorMessage } from "@/errors/errorHelpers";
...
   onSelect,
 }) => {
-  const { data, error } = useAsyncState(
-    async () =>
-      queryTabs({
-        url: "*://*/*",
-      }),
-    [],
-  );
+  const { data, error } = useAsyncState(async () => queryTabs({}), []);
```
Comment (mnholtz) [inline src/pageEditor/context/TabInspectionGate.tsx:34]: Broadening the tabs that you can open the Page Editor for here in order to be able to test restricted browser pages. I personally don't see a problem with this since this is an internal-use feature, but let me know if there is anything out of my frame of context that might affect that opinion.

File: end-to-end-tests/tests/pageEditor/specialPages.spec.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  const modId = "@e2e-testing/googlecom-trigger";
+  const modActivationPage = new ActivateModPage(page, extensionId, modId);
+  await modActivationPage.goto();
+  await modActivationPage.clickActivateAndWaitForModsPageRedirect();
+
+  await page.goto("/");
+  const pageEditorPage = await newPageEditorPage(page.url());
+  await pageEditorPage.modListingPanel
+    .getModListItemByName("Google.com trigger")
+    .click();
```
Comment (fungairino) [inline end-to-end-tests/tests/pageEditor/specialPages.spec.ts:53]: nit: this is "selecting" the modListItem, and we have a method to do that on the modListItem pom. Maybe we can just remove that since it just calls through to click anyways.

File: end-to-end-tests/tests/pageEditor/specialPages.spec.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    pageEditorPage.modListingPanel.getModStarterBrick(
+      "Google.com trigger",
+      "Google.com trigger",
+    );
+
+  await expect(
+    googleTriggerStarterBrick.getByRole("img", {
+      name: "Not available on page",
+    }),
+  ).toBeVisible();
```
Comment (fungairino) [inline end-to-end-tests/tests/pageEditor/specialPages.spec.ts:64]: nit: for readability you could consider extracting this locator to the mod list item pom.

File: end-to-end-tests/tests/pageEditor/specialPages.spec.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    await modActivationPage.clickActivateAndWaitForModsPageRedirect();
+  });
+
+  await test.step("Verify the add brick modal is hidden after the Page Editor refreshes", async () => {
+    await expect(
+      pageEditorPage.getByText(
+        "There were changes made in a different instance of the Page Editor. Reload this Page Editor to sync the changes.",
+      ),
+    ).toBeVisible();
+  });
```
Comment (fungairino) [inline end-to-end-tests/tests/pageEditor/specialPages.spec.ts:93]: What about editing the same mod in another page editor instance? I think in that case we show the same message. Might be good to add as another case to test in this flow.

File: src/pageEditor/context/TabInspectionGate.tsx (hunk 1 above)
Comment (fungairino) [reply src/pageEditor/context/TabInspectionGate.tsx:34]: You may also consider navigating to the restricted page in the connected page in the e2e test.

File: end-to-end-tests/tests/pageEditor/specialPages.spec.ts (hunk 4 above)
Comment (mnholtz) [reply end-to-end-tests/tests/pageEditor/specialPages.spec.ts:93]: Great idea! I'll take that approach

File: src/store/sessionChanges/sessionChangesListenerMiddleware.ts [hunk 5]
```diff
@@ -35,6 +35,7 @@ sessionChangesListenerMiddleware.startListening({
     actions.editModOptionsValues,
     actions.resetMetadataAndOptionsForMod,
     actions.addModComponentFormStateToMod,
+    actions.addModComponentFormState,
```
Comment (mnholtz) [inline src/store/sessionChanges/sessionChangesListenerMiddleware.ts:38]: See conversation here: https://pixiebrix.slack.com/archives/C0436P48QHY/p1721080156329249

It looks like we weren't invalidating the page editor session on when adding a new mod component (i.e. adding a new starter brick). This is the fix, confirmed by my third test I've introduced in this PR

File: end-to-end-tests/tests/pageEditor/specialPages.spec.ts (hunk 2 above)
Comment (mnholtz) [reply end-to-end-tests/tests/pageEditor/specialPages.spec.ts:54]: Let's discuss removing later... touches on the "selected/active" terminology conversation we've been having

File: end-to-end-tests/pageObjects/pageEditor/modListingPanel.ts [hunk 6]
```diff
@@ -32,6 +32,12 @@ export class ModListItem extends BasePageObject {
     return this.getByLabel(" - Ellipsis");
   }
 
+  async isAvailable() {
```
Comment (fungairino) [inline end-to-end-tests/pageObjects/pageEditor/modListingPanel.ts:35]: generally I prefer to keep simple locators on the pom, and then perform the kind of action you want to do on it in the actual test. If you need to perform more than one action, or do some condition, then I think it makes sense to refactor into a helper method.

**Review by fungairino — APPROVED** [review]

**Comment by twschiller** [comment]
> Note that it also broadens the tab query in TabInspectionGate.tsx (the component that renders the list of tabs to open in the tab-version of the Page Editor) to include restricted browser pages (in order to enable testing restricted browser page behavior in the page edtior)

You might choose to only include those in debug builds. Or leave a reminder/note to remove that if we ever move to support the Page Editor outside of the dev tools

File: end-to-end-tests/pageObjects/pageEditor/modListingPanel.ts (hunk 6 above)
Comment (mnholtz) [reply end-to-end-tests/pageObjects/pageEditor/modListingPanel.ts]: ok

**Comment by mnholtz** [comment]
> You might choose to only include those in debug builds. Or leave a reminder/note to support excluding those that if we ever move to support the Page Editor outside of the dev tools

@twschiller Forgot to update that comment; I took it down in favor of navigating to the extension console after connecting the page editor to a tab
